move math constants to standard library

amtoine opened the pull request:
# Description
we talked about this before in some meetings so i thought, why not?

the hope is that these constants do not require Rust code to be implemented and that this move will make the Rust source base a bit smaller :crossed_fingers: 

# User-Facing Changes
mathematical constants (e, pi, tau, phi and gamma) are now in `std math` rather than `math`

## what can be done
```nushell
> use std; $std.math
> use std math; $math
> use std *; $math
```
will all give
```
╭───────┬────────────────────╮
│ GAMMA │ 0.5772156649015329 │
│ E     │ 2.718281828459045  │
│ PI    │ 3.141592653589793  │
│ TAU   │ 6.283185307179586  │
│ PHI   │ 1.618033988749895  │
╰───────┴────────────────────╯
```
and the following will work too
```nushell
> use std math E; $E
2.718281828459045
```
```nushell
> use std math *; $GAMMA
0.5772156649015329
```

## what can NOT be done
looks like every export works fine now :relieved: 

# Tests + Formatting
# After Submitting

jntrnr commented on crates/nu-std/std/math.nu:
```diff
@@ -0,0 +1,22 @@
+# return the Euler-Mascheroni constant
+#
+# see https://en.wikipedia.org/wiki/Euler%27s_constant
+export def gamma [] { 0.5772156649015329 }
```
feels like these should be `export const` but not sure if we support that yet

amtoine replied:
very good catch, i think we do now :open_mouth:

amtoine replied:
done in 57da75580

amtoine replied:
i've updated the description of the PR with the possible usage :+1: 

> **Note**
> `use std math; $math` does not work, but i think it should :thinking:

amtoine commented:
marking as DRAFT until the CI passes... :recycle:

sholderbach commented on crates/nu-cmd-extra/src/extra/math/cos.rs:
```diff
@@ -54,7 +54,7 @@ impl Command for SubCommand {
             Example {
                 description: "Apply the cosine to π",
-                example: "math pi | math cos",
+                example: "3.141592 | math cos | math round --precision 4",
```
Grr, I see why it is necessary but I dislike the impact on the educational value of those examples, when they get stuffed with evermore unrelated parts to let them work as tests.

amtoine replied:
agree :thinking:

amtoine replied:
@sholderbach
is there a way to allow these example tests to `use std math PI`?

amtoine replied:
also we'd need the use of `math round` because the precision of Nushell floats is not enough for commands like `math sin` to evaluate `PI` to integers :eyes:

kubouch commented:
I think this PR is blocked on the fact that constant exporting does not work recursively. https://github.com/nushell/nushell/pull/9773 only exports const from a top-level module but does not go to submodules.

amtoine commented:
let's wait for https://github.com/nushell/nushell/pull/10049 to land then :relieved:

kubouch commented:
@amtoine Should be unblocked now

amtoine commented:
@kubouch 
i've updated the PR, tested again and updated the PR description :+1: 
there are good things but still some other unexpected commands that do not work apparently :thinking:

amtoine commented:
just making sure this is expected before moving on to the tests

sholderbach commented:
What is the status on this one since consts can be exported from modules?

amtoine commented:
> What is the status on this one since consts can be exported from modules?
>
might have forgotten to mark that as ready for review, my bad :pray: 

if we're ok with the examples in the PR description and the open thread above, then let's go :yum:

sholderbach commented:
Let's give it a go
